### PR TITLE
Create a database based on a diagram

### PR DESCRIPTION
**Pull request by egichuhis**
Hello, please review this PR.

The Diagram:
clinic_diagram

Tables added:

1. patients
2. medical_histories
3. treatments
4. invoices
5. invoice_items

To establish the correct relationships between the tables, we have added the necessary foreign key constraints and indexes.

patients
medical_histories
treatments
invoices
invoice_items

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+CREATE INDEX ON "medical_histories"
+("patient_id");
+
+CREATE TABLE "treatments"
+(
+  "id" SERIAL PRIMARY KEY,
+  "type" VARCHAR(255),
+  "name" VARCHAR(255)
+);
```
Comment (mckent05) [inline schema_based_on_diagram.sql:25]: - You have done well so far, but you are yet to create a `many-many` relationship between the `medical_histories` table and the `treatments` table. You can implement this by creating a join table, `medical_histories_treatments`. Please fix this 😄 
- Do not forget to add indexes on the foreign keys for the join table, as required by the project requirements 👍 
```SQL  
CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @egichuhis, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- You have done well creating indexes on all foreign keys ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job creating your database using the given ERD 👍 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (egichuhis) [reply schema_based_on_diagram.sql:25]: Thanks @mckent05 for noting this. 😄

**Review by V-Blaze — APPROVED** [review]
Hi @egichuhis, @evansnyamekye  

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

 ## Highlights
- All tables created correctly ✔
-  "join table" for the many-to-many relationship is created👋 
-  Correct git flow✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**